www/include/configuration/configKnowledge/display-hosts.php

```diff
@@ -104,45 +103,55 @@
 
     $query = "SELECT SQL_CALC_FOUND_ROWS host_name, host_id, host_register, ehi_icon_image " .
         "FROM extended_host_information ehi, host ";
-    if (isset($_REQUEST['searchPoller']) && $_REQUEST['searchPoller']) {
-        $query .= " JOIN ns_host_relation nhr ON nhr.host_host_id = host.host_id ";
+    if ($postPoller !== false) {
+        $query .= "JOIN ns_host_relation nhr ON nhr.host_host_id = host.host_id ";
+    }
+    if ($postHostgroup !== false) {
+        $query .= "JOIN hostgroup_relation hgr ON hgr.host_host_id = host.host_id ";
+    }
+    $query .= "WHERE host.host_id = ehi.host_host_id ";
+    if ($postPoller !== false) {
+        $query .= "AND nhr.nagios_server_id = :postPoller ";
+    }
+    $query .= "AND host.host_register = '1' ";
+    if ($postHostgroup !== false) {
+        $query .= "AND hgr.hostgroup_hg_id = :postHostgroup ";
     }
-    if (isset($_REQUEST['searchHostgroup']) && $_REQUEST['searchHostgroup']) {
-        $query .= " JOIN hostgroup_relation hgr ON hgr.host_host_id = host.host_id ";
+    if (!empty($postHost)) {
+        $query .= "AND host_name LIKE :postHost ";
     }
-    $query .= " WHERE host.host_id = ehi.host_host_id ";
-    if (isset($_REQUEST['searchPoller']) && $_REQUEST['searchPoller']) {
-        $query .= " AND nhr.nagios_server_id = " . $pearDB->escape($_REQUEST['searchPoller']);
+    $query .= "ORDER BY " . $orderby . " " . $order . " LIMIT " . $num * $limit . ", " . $limit;
+    $statement = $pearDB->prepare($query);
+    if ($postPoller !== false) {
```

Comment:
you should use a $queryValue array in a foreach loop, to avoid to check again what is set or not